openstack/clustering/v1/policies/results.go

```diff
@@ -54,21 +56,72 @@ func (r PolicyPage) LastMarker() (string, error) {
 	return policies[len(policies)-1].ID, nil
 }
 
+const RFC3339WithZ = "2006-01-02T15:04:05Z"
+
 func (r *Policy) UnmarshalJSON(b []byte) error {
 	type tmp Policy
 	var s struct {
 		tmp
-		CreatedAt gophercloud.JSONRFC3339MilliNoZ `json:"created_at,omitempty"`
-		UpdatedAt gophercloud.JSONRFC3339MilliNoZ `json:"updated_at,omitempty"`
+		CreatedAt string `json:"created_at,omitempty"`
+		UpdatedAt string `json:"updated_at,omitempty"`
 	}
 	err := json.Unmarshal(b, &s)
 	if err != nil {
 		return err
 	}
 	*r = Policy(s.tmp)
 
-	r.CreatedAt = time.Time(s.CreatedAt)
-	r.UpdatedAt = time.Time(s.UpdatedAt)
+	if s.CreatedAt != "" {
+		r.CreatedAt, err = time.Parse(gophercloud.RFC3339MilliNoZ, s.CreatedAt)
+		if err != nil {
+			r.CreatedAt, err = time.Parse(RFC3339WithZ, s.CreatedAt)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	if s.UpdatedAt != "" {
+		r.UpdatedAt, err = time.Parse(gophercloud.RFC3339MilliNoZ, s.UpdatedAt)
+		if err != nil {
+			r.UpdatedAt, err = time.Parse(RFC3339WithZ, s.UpdatedAt)
+			if err != nil {
+				return err
+			}
+		}
+	}
 
 	return nil
 }
+
+type policyResult struct {
+	gophercloud.Result
+}
+
+func (r CreateResult) Extract() (*Policy, error) {
+	var s struct {
+		Policy *Policy `json:"policy"`
+	}
+	err := r.ExtractInto(&s)
+
+	return s.Policy, err
+}
+
+type CreateResult struct {
+	policyResult
+}
+
+type PolicyDeleteResult struct {
+	gophercloud.HeaderResult
+}
+
+// DeleteResult contains the delete information from a delete policy request
+type DeleteResult struct {
```

Comment:
Let's call this `DeleteHeader`. There's precedence for this name here: https://github.com/gophercloud/gophercloud/blob/master/openstack/objectstorage/v1/containers/results.go#L294-L301